backend/internal/moduleindex/generator.go

```diff
@@ -593,9 +594,15 @@ func (g generator) extractSubmodules(ctx context.Context, addr ModuleAddr, ver M
 				Resources:    []Resource{},
 			},
 		}
-		submodulePrefix := path.Join(directoryPrefix, name)
-		if err := g.refreshModuleDetails(ctx, addr, ver, &submodule.Details, workingCopy, licenseOK, blocked, blockedReason, submodulePrefix); err != nil {
-			return fmt.Errorf("failed to refresh details for submodule %s (%w)", submodulePrefix, err)
+		sourcePrefix := path.Join(directoryPrefix, name)
+		dstPrefix, err := g.sanitizePath(directoryPrefix, name)
+		if err != nil {
+			// Invalid path, ignore.
+			g.log.Warn(ctx, "Failed to index %s version %s submodule %s (%v)", addr, ver.ID, dstPrefix, err)
+			continue
+		}
+		if err := g.refreshModuleDetails(ctx, addr, ver, &submodule.Details, workingCopy, licenseOK, blocked, blockedReason, sourcePrefix, dstPrefix); err != nil {
```

Comment:
nit: makes sense to put all those arguments into a struct, something like `ModuleDetailsRefreshOptions` to reduce the overall number of arguments